src/v/serde/parquet/column_writer.cc

```diff
@@ -89,7 +89,7 @@ class buffered_column_writer final : public column_writer::impl {
                   value_memory_usage = sizeof(value_type);
```

Comment:
I wouldn't expect this change to move the needle much for the primitive types, but I thought something like this would help a ton for the `iobuf`/byte-array case, since we remove the accumulation into the intermediate `chunked_vector<iobuf>` entirely which removes a lot of the iobuf manipulation.

The overall results seem really bad: about 300 instructions to do the write flow for a single primitive value. Is compression included in the measured region? How much is compression alone?

Reply:
> results in fewer allocations and slightly better performance.

The allocations look almost identical in both cases? Are you talking about a reduction from like `1.084` to `1.074`? That's pretty minuscule, just 1% of this microbench.

Reply:
Note that the values in the microbenchmark results normalized to be per byte added(they were per value before  my latest push).